File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const map = new Map();
    
    for (let cloth of clothes) {
        map.set(cloth[1], (map.get(cloth[1])||0)+1);
    }
    
    let count = 1;
    for (let value of map.values()) {
        count *= value+1;
    }
    
    return count-1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    hm = {}
    for cloth in clothes:
        if cloth[1] in hm:
            hm[cloth[1]]+=1
        else:
            hm[cloth[1]]=1
            
    answer = 1
    for h in hm.items():
        answer*=h[1]+1
    
    return answer-1